### PR TITLE
Update sync logic to ignore unconnected peers

### PR DESCRIPTION
**Pull request by JimmyShi22**

### CODE REVIEW

File: libsync/SyncMaster.cpp [hunk 1]
```diff
@@ -484,13 +484,25 @@ bool SyncMaster::maintainDownloadingQueue()
...
     NodeIDs miners = m_blockChain->minerList();
-    NodeIDs groupMembers = miners + m_blockChain->observerList();
+    NodeIDs minerOrObserver = miners + m_blockChain->observerList();
 
-    // Delete uncorrelated peers
+    // member set is [(miner || observer) && activePeer && not myself]
     set<NodeID> memberSet;
-    for (auto member : groupMembers)
-        memberSet.insert(member);
+    for (auto member : minerOrObserver)
```
Comment (wheatli) [inline libsync/SyncMaster.cpp]: auto const &

File: libsync/SyncMaster.cpp [hunk 2]
```diff
@@ -534,10 +546,10 @@ void SyncMaster::maintainPeersConnection()
     });
 
     // If myself is not in group, ignore receive packet checking from all peers
-    if (memberSet.find(m_nodeId) == memberSet.end())
-        m_msgEngine->needCheckPacketInGroup = false;
-    else
-        m_msgEngine->needCheckPacketInGroup = true;
+    bool hasMyself = false;
+    for (auto const& mo : minerOrObserver)
```
Comment (wheatli) [inline libsync/SyncMaster.cpp]: do you know there is a function named exist defined in map/set?

Comment (wheatli) [reply libsync/SyncMaster.cpp]: Sorry, I thought the minerOrObserver is a map or set.
What I think is that you don't need to traverse the list again, may be you can figure it out when you traversed last time

File: libsync/SyncMaster.cpp (hunk 1 above)
Comment (JimmyShi22) [reply libsync/SyncMaster.cpp]: fixed

File: libsync/SyncMaster.cpp (hunk 2 above)
Comment (JimmyShi22) [reply libsync/SyncMaster.cpp]: Use find instead. About map/set, to detected existing, maybe also use find() == end()

Comment (JimmyShi22) [reply libsync/SyncMaster.cpp]: Sure, fix it immediately

**Pull request merged**